tests/Commands/CreateOrEditTaskParser.test.ts

```diff
@@ -96,3 +97,40 @@ describe('CreateOrEditTaskParser - task recognition', () => {
         expect(task.doneDate).toEqualMoment(moment('2023-06-22'));
     });
 });
+
+describe('CreateOrEditTaskParser - created date', () => {
+    beforeAll(() => {
+        jest.useFakeTimers();
+        jest.setSystemTime(new Date('2023-09-17'));
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+    });
+
+    it.each([
+        ['- ', '- [ ]  ➕ 2023-09-17', '2023-09-17'], // bullet point only
+        ['- [ ] ', '- [ ]  ➕ 2023-09-17', '2023-09-17'], // bullet point and a checkbox
+        [
+            '- [ ] without global filter and without created date', // nominal use-case
+            '- [ ] without global filter and without created date ➕ 2023-09-17',
```

Comment:
Could you please say why this one adds created date? I don't think either of the referenced Issues requests this.

- One of them talks about if the global filter was absent.
-  The other talks about if the description was empty

Reply:
What am I missing?

Reply:
Because if we use the Create or Edit Task command on such a task line, it shall become a task, treated by Tasks plugin. Since the created date option is on, it is added to the line.

Pls lmk if this reasoning is wrong.

Reply:
Because if we use the Create or Edit Task command on such a task line, it shall become a task, treated by Tasks plugin. Since the created date option is on, it is added to the line.

Pls lmk if this reasoning is wrong.

Reply:
Because if we use the Create or Edit Task command on such a task line, it shall become a task, treated by Tasks plugin. Since the created date option is on, it is added to the line.

Pls lmk if this reasoning is wrong.

Reply:
Because if we use the Create or Edit Task command on such a task line, it shall become a task, treated by Tasks plugin. Since the created date option is on, it is added to the line.

Pls lmk if this reasoning is wrong.

Reply:
I believe this is the case described in #2112

Reply:
I have about a thousand tasks that I created before the Created date was added. If I understand this behaviour, every time I edit any of them, like to change the Due date, they will be marked with Today’s date. That is not what was intended.

Reply:
> I believe this is the case described in #2112

No, that one specifically only applies to the case where the user settings have a global filter and the initial task line did not have a global filter. So the previous line did not count as a task because of the missing global filter, so the edit that adds the global filter also adds the Created date.

Reply:
I was puzzled about the logic of fixing both issues in the same PR as they seemed quite different logic to me. Probably the misunderstanding about the global filter explains why it seemed a good idea to both at once.

Reply:
Many thanks for catching this. I fixed this behaviour.

It seems to me that the test I wrote were expressive enough to be understood. This makes me happy =)

Reply:
Yes, I was able to understand the behaviour from the tests which was great.

Thanks for fixing this.

> fix: created date not automatically added when the line has item marker and checkbox and/or more data

I think this PR description is now wrong??? the `and/or more data` bit...

Reply:
oops, I fixed the description but forgot about the title indeed, thanks!